### ISSUE TITLE
Discuss including example of jupyterlab distribution in core

### ISSUE THREAD

**Issue by saulshanabrook** [jupyterlab/jupyterlab#6090]
@jonmmease worked on a nice example of creating a custom JupyterLab "distribution": https://github.com/jonmmease/jupyterlab_delux

Why? So that when a user conda installs this package, they don't have to have JS installed or rebuild jupyterlab to get started with a number of packages.

It would be good to reference this approach at least from our docs, and we could discuss seeing if `jupyterlab_delux` wanted to come into the core `jupyterlab` org as another cookiecutter repo.

**Comment by bollwyvl** [comment]
Indeed, for a workshop we went a bit further to generate a full offline-installable [python+conda+kernels+lab+libraries+examples](https://github.com/robots-from-jupyter/robotlab
) distribution.

On the _other_ end of the spectrum, on some projects I've also considered checking into VCS a built custom lab, with a wrapper script to launch... this would let a project build the app once (or in ci), and lab _users_ don't have to worry about node.

I think if we want to make the multiple apps approach pallatable, in _addition_ to documentation, there are a lot of affordances we might want to consider.

Introduce a **minimal UI** for discovering and switching between lab-adjacent UX, 
  - _without_ restarting the server (not really an option in a binder-like setting)
  - for lab-adjacent apps, show lab, classic, nteract, graphql (from metadata), rstudio & theia (from jupyter-server-proxy), ...
    - show useful metadata from...

...An **appspec** (like kernelspec) directory structure
  - could just be a `package.json`, but can't assume `node`...
  -  the usual suspects, e.g. label, version, description, license, and support url
  - but maybe include some more, specifically geared to UX:
   - icon, docs, accessibility, internationalization
  - make it easy to get the `package.json` out of...

...An archive format, e.g. `foo.v0.lab.app.zip`
  - if unmodified, just serve them on-demand from a temp dir
  - _if_ the user has `node`, installing new extensions _would_ trigger an unpacking into a normal directory structure, but the archive would be maintained to have a "last known good" state
  -  The butcher's bill for `0.35.4` core as-served is 26MB in unzipped dev assets, which gzips down to 7MB, or 3.7M with 7z/br. If users don't care about source maps, a zip gets down to 4.3MB, with 7z to 2.4MB.
  - which can be predictably found in...

...Some **well-known places** where users and package managers can easily put built apps (whether in servable or archive form) in a namespaced kinda way:
  - `$PREFIX/share/jupyter/lab/apps/foo-lab-v0`
  - `$(cwd)/.jupyter/lab/apps/bar-lab-v1.br`

**Comment by jhermann** [comment]
FYI, JupyterLab is a to-do for https://github.com/1and1/debianized-jupyterhub, which is a turn-key solution for the Debian-verse.
